crates/ruff_server/src/server/api/requests/code_action.rs

```diff
@@ -56,21 +67,22 @@ impl super::BackgroundDocumentRequestHandler for CodeActions {
 
 fn quick_fix(
     snapshot: &DocumentSnapshot,
-    diagnostics: Vec<types::Diagnostic>,
+    fixes: &[DiagnosticFix],
 ) -> crate::Result<Vec<CodeActionOrCommand>> {
     let document = snapshot.document();
-
-    let fixes = fixes_for_diagnostics(document, snapshot.encoding(), diagnostics)?;
-
     fixes
-        .into_iter()
+        .iter()
+        .filter(|fix| fix.edits.is_some())
         .map(|fix| {
             let mut tracker = WorkspaceEditTracker::new(snapshot.resolved_client_capabilities());
 
             tracker.set_edits_for_document(
                 snapshot.url().clone(),
                 document.version(),
-                fix.edits,
+                fix.edits
+                    .as_ref()
+                    .expect("should only be iterating over fixes with available edits")
+                    .clone(),
             )?;
 
```

Comment:
I would use `filter_map` to avoid the `expect` call
```suggestion
        .filter_map(|fix| {
            let edits = fix.edits.as_ref()?;

            let mut tracker = WorkspaceEditTracker::new(snapshot.resolved_client_capabilities());

            tracker.set_edits_for_document(
                snapshot.url().clone(),
                document.version(),
                edits
            )?;

```